### DISCUSSION TITLE
Is there any less privileged role than manage-users for realm-management client.?

### DISCUSSION THREAD

**Discussion by amolai** [keycloak/keycloak#33507] [Archive]
Disclaimer: Not completely sure which category this query/discussion would belong to hence adding it here, kindly correct me or move the conversation to the right place

Problem Description: We have created a custom user role in our realm by name custom user, this role has a realm-management permission/role view-users. We have a user settings UI screen, and we want the user to be able to update their own password, email, firstName etc from this screen. This is not possible through only one permission view-users, when we add manage-users role it works fine. But the downside here is that with manage-users, this user can escalate its privileges to another powerful user role like custom admin which we have defined and acts like a super user. 

Is there any way to achieve this? I have tried adding account level permissions like manage-account, view-profile etc. But without  manage-users nothing works for me. 

Any guidance here would be helpful. 

Thanks in advance.

**Comment by codespearhead** [comment]
> We have a user settings UI screen, and we want the user to be able to update their own password, email, firstName etc from this screen

Is there a reason why the built-in Account Console [1] doesn't meet your needs?

Alternatively, you could create a custom role called `manage-own-profile`, which allows users to manage certain aspects of their own profile using fine-grained admin permissions [2]. Just keep in mind that this is still a preview feature.

[1] https://www.keycloak.org/docs/latest/server_admin/#accessing-the-account-console
[2] https://www.keycloak.org/docs/latest/server_admin/#_fine_grain_permissions

**Reply by amolai** [reply]
@codespearhead Thanks for the response, that was helpful. Our requirement is that we want to provide a screen which allows user to update the profile as well as the password. Until now we were using the admin apis for the same but that required manage-users permission which cannot be given to a user. 

1. For updating user profile email, firstname etc we could use the account API the way it is being used in the account-console.
2. For password change as I explored, there is no api available. In the account-console as I can see for changing the password execute and required-action are being used with redirection, is it possible to achieve this using API calls ? 

Any input help here would be helpful, thanks

**Reply by codespearhead** [reply]
Look up the term "reset-password" (prefer using the "reset-password-email" endpoint for security reasons) in the Keycloak Admin Rest API [1].

[1] https://www.keycloak.org/docs-api/latest/rest-api/index.html

**Reply by amolai** [reply]
@codespearhead For using reset-password or reset-password-email, manage-users role is needed and we cannot assign this. Any api we can use from Account console to update the password? As we want to provide a screen to the user to update password

Ideally reset-password should be only available to admin.
